lib/mapnik_backend.js

```diff
@@ -175,29 +166,17 @@ MapnikSource.prototype._open = function(uri) {
 
     // Initialize this map. This wraps `localize()` and calls `create()`
     // afterwards to actually create a new Mapnik map object.
-    Step(function() {
-        source._loadXML(this);
-    }, function(err, xml) {
-        if (err) return error(err);
-        // https://github.com/mapbox/tilelive-mapnik/issues/25
-        // there seems to be no value to assinging xml to a property
-        //source._xml = xml;
-        source._createMetatileCache(uri.query.metatileCache);
-        source._createPool(xml, this);
-    }, function(err) {
-        if (err) return error(err);
-        source._populateInfo(this);
-    }, function(err) {
-        if (err) return error(err);
-        source.open = true;
-        source.emit('open', null, source);
+    this._loadXML((err, xml) => {
+        if (err) return this.emit('open', err);
+
+        this._createMetatileCache(uri.query.metatileCache);
+        this._createPool(xml);
+        this._populateInfo();
```

Comment:
Just curious, why do you move `this.open = true` and `this.emit('open', null, this)` to `._populateInfo()`?

Reply:
In insight it was probably an error since it was clearer having it in _open. I'm getting kind of tired of mixing events for callback so I'll try to remove them and see how it gets.

Reply:
Done in 561800e.